### ISSUE TITLE
Values that begin with a hyphen are not handled properly

### ISSUE THREAD

**Issue by bmw** [bw2/ConfigArgParse#154]
In my project that uses configargparse, we have the somewhat strange use case of needing to have command line flags that accept values starting with a hyphen like this:
```
--my-flag -my-value
```
See https://github.com/certbot/certbot/issues/7114.

The normal workaround for this suggested at https://bugs.python.org/issue9334 for this when using argparse is to separate the flag and value with an equals sign so the command line becomes:
```
--my-flag=-my-value
```
This works with `argparse`:
```
>>> import argparse
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--my-flag')
_StoreAction(option_strings=['--my-flag'], dest='my_flag', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> parser.parse_args(['--my-flag=-my-value'])
Namespace(my_flag='-my-value')
```
Unfortunately, this doesn't work with configargparse. Using Python 3.7.3 and ConfigArgParse 0.14.0, I saw this behavior:
```
>>> import configargparse
>>> parser = configargparse.ArgumentParser()
>>> parser.add_argument('--my-flag')
_StoreAction(option_strings=['--my-flag'], dest='my_flag', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> parser.parse_args(['--my-flag=-my-value'])
usage: [-h] [--my-flag MY_FLAG]
: error: argument --my-flag: expected one argument
```
In [this comment](https://github.com/certbot/certbot/issues/7114#issuecomment-507535101) someone bisected this project and said that the this problem was introduced in https://github.com/bw2/ConfigArgParse/commit/e0c35f9347bc19091f51c4c2251c650bcf6735f8.

It would help us a lot of configargparse had the same behavior as argparse here and this problem could be resolved.

**Comment by bw2** [comment]
A pull request with a test + fix for this issue would be appreciated.

**Comment by bw2** [comment]
This has been fixed by PR #166 and pushed to PyPI.

**Comment by bmw** [comment]
Thank you!
